### PR TITLE
Add JSON support for brew cask outdated, fixes #7432

### LINKED ISSUE #7432: Support json output for `brew cask outdated`
# A detailed description of the proposed feature

I would like to get the output of `brew cask outdated` in json format. eg: 
```
brew cask outdated --json
```

# The motivation for the feature
It would be great to have `brew cask outdated` support the similar features of `brew outdated`.

# How the feature would be relevant to at least 90% of Homebrew users
Since `brew outdated --json` provides a json output, to help tools that depend on json output to have common logic.

### PR DESCRIPTION
**Pull request by FTLam11**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----

This PR adds JSON support to `brew cask outdated`, including this change gives more feature parity with `brew outdated` (which already has JSON support). This PR addresses issue #7432

### CODE REVIEW

File: Library/Homebrew/cask/cask.rb [hunk 1]
```diff
@@ -122,6 +122,21 @@ def outdated_versions(greedy = false)
       installed.reject { |v| v == version }
     end
 
+    def outdated_info(greedy, verbose, json)
+      if json
+        {
+          name:               token,
+          installed_versions: outdated_versions(greedy).join(", "),
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/cask.rb:129]: Thoughts about splitting `outdated_versions(greedy).join(", ")` into a variable?

Comment (FTLam11) [reply Library/Homebrew/cask/cask.rb]: Done! I also moved the --quiet case to its own early return clause.

File: Library/Homebrew/cask/cask.rb [hunk 2]
```diff
@@ -122,6 +122,22 @@ def outdated_versions(greedy = false)
       installed.reject { |v| v == version }
     end
 
+    def outdated_info(greedy, verbose, json)
+      return token unless verbose || json
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/cask.rb:126]:
```suggestion
      return token if !verbose && !json
```

is a bit easier to follow

**Comment by MikeMcQuaid** [comment]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @FTLam11!

**Pull request merged**